### ISSUE TITLE
TXT records can no longer have empty text

### ISSUE THREAD

**Issue by johan13** [mafintosh/dns-packet#39]
Starting with commit f6db3d3d, I can no longer send TXT records with an empty payload using [multicast-dns](https://www.npmjs.com/package/multicast-dns):
```javascript
{name, type: "TXT", ttl, data: ""}
```

I have identified the problem to be the coalescing of data and options on line 1241 of dns-packet:
https://github.com/mafintosh/dns-packet/blob/0f87ff09f523447165848086534c477ca51337fd/index.js#L1241

Would it be possible to change that into a comparison with `null` and/or `undefined` instead to allow empty TXT records again? Sending empty TXT records may seem pointless, but I found it to be required for interoperability with certain mDNS clients.

**Comment by johan13** [comment]
I found a workaround by passing `Buffer.alloc(0)` instead of `""`, but maybe it'd be nice to get this fixed anyway.

**Comment by silverwind** [comment]
With "send"  you mean encoding? Maybe I'm missing something but those inputs seem to encode the same here:

````js
> require(".").encode({type: 'response', answers: [{name: ".", type: 'TXT',data: ''}]})
<Buffer 00 00 80 00 00 00 00 01 00 00 00 00 00 00 10 00 01 00 00 00 00 00 01 00>
> require(".").encode({type: 'response', answers: [{name: ".", type: 'TXT',data: Buffer.alloc(0)}]})
<Buffer 00 00 80 00 00 00 00 01 00 00 00 00 00 00 10 00 01 00 00 00 00 00 01 00>
> require(".").encode({type: 'response', answers: [{name: ".", type: 'TXT',data: ['']}]})
<Buffer 00 00 80 00 00 00 00 01 00 00 00 00 00 00 10 00 01 00 00 00 00 00 01 00>
````

**Comment by johan13** [comment]
That looks good, but I get a `TypeError` when I run the same code:
```
$ git clone https://github.com/mafintosh/dns-packet.git ; cd dns-packet ; npm i ; node -e "require('.').encode({type: 'response', answers: [{name: '.', type: 'TXT',data: ''}]})"
Cloning into 'dns-packet'...
remote: Counting objects: 301, done.
remote: Compressing objects: 100% (15/15), done.
remote: Total 301 (delta 10), reused 17 (delta 6), pack-reused 279
Receiving objects: 100% (301/301), 112.83 KiB | 560.00 KiB/s, done.
Resolving deltas: 100% (171/171), done.
npm notice created a lockfile as package-lock.json. You should commit this file.
added 202 packages in 3.991s
/Users/jlevin/Projects/dns-packet/index.js:350
      length += buf.length + 1
                    ^

TypeError: Cannot read property 'length' of undefined
    at /Users/jlevin/Projects/dns-packet/index.js:350:21
    at Array.forEach (<anonymous>)
    at Object.rtxt.encodingLength (/Users/jlevin/Projects/dns-packet/index.js:346:8)
    at Object.answer.encodingLength (/Users/jlevin/Projects/dns-packet/index.js:1241:57)
    at encodingLengthList (/Users/jlevin/Projects/dns-packet/index.js:1382:52)
    at Object.exports.encodingLength (/Users/jlevin/Projects/dns-packet/index.js:1351:5)
    at Object.exports.encode (/Users/jlevin/Projects/dns-packet/index.js:1304:46)
    at [eval]:1:14
    at ContextifyScript.Script.runInThisContext (vm.js:50:33)
    at Object.runInThisContext (vm.js:139:38)
```

I run this on a Mac with Node 8:
```
$ node -p process.versions
{ http_parser: '2.8.0',
  node: '8.11.3',
  v8: '6.2.414.54',
  uv: '1.19.1',
  zlib: '1.2.11',
  ares: '1.10.1-DEV',
  modules: '57',
  nghttp2: '1.32.0',
  napi: '3',
  openssl: '1.0.2o',
  icu: '62.1',
  unicode: '11.0',
  cldr: '33.1',
  tz: '2018e' }
```

**Comment by silverwind** [comment]
Yeah, I was actually playing around earlier with a fix, so I didn't notice. Fixed in dns-packet@5.0.1.

**Comment by johan13** [comment]
👍 Thank you!
